src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/VsLangProjectPropertiesTests.cs

```diff
@@ -0,0 +1,160 @@
+
+using System;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Moq;
+using VSLangProj;
+using VSLangProj110;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ProjectSystemTrait]
+    public class VsLangProjectPropertiesTests
+    {
+        [ProjectSystemTrait]
+        public class VsLangProjectPropertiesProviderTests
+        {
+            [Fact]
+            public void Constructor_NullAsVsProject_ThrowsArgumentNull()
+            {
+                Assert.Throws<ArgumentNullException>("vsProject", () =>
+                {
+                    GetVsLangProjectProperties();
+                });
+            }
+
+            [Fact]
+            public void Constructor_NullAsThreadingService_ThrowsArgumentNull()
+            {
+                Assert.Throws<ArgumentNullException>("threadingService", () =>
+                {
+                    GetVsLangProjectProperties(Mock.Of<VSProject>());
+                });
+            }
+
+            [Fact]
+            public void Constructor_NullAsProjectProperties_ThrowsArgumentNull()
+            {
+                Assert.Throws<ArgumentNullException>("projectProperties", () =>
+                {
+                    GetVsLangProjectProperties(Mock.Of<VSProject>(), Mock.Of<IProjectThreadingService>());
+                });
+            }
+
+            [Fact]
+            public void VsLangProjectProperties_NotNull()
+            {
+                var provider = GetVsLangProjectProperties(Mock.Of<VSProject>(), Mock.Of<IProjectThreadingService>(), Mock.Of<ActiveConfiguredProject<ProjectProperties>>());
+                Assert.NotNull(provider);
+            }
+
+            [Fact]
+            public void VsLangProjectProperties_OutputTypeEx_Get()
+            {
+                var project = UnconfiguredProjectFactory.Create();
+                var enumValue = new Mock<IEnumValue>();
+                enumValue.Setup(s => s.DisplayName).Returns("2");
+                var data = new PropertyPageData()
+                {
+                    Category = ConfigurationGeneralBrowseObject.SchemaName,
+                    PropertyName = ConfigurationGeneralBrowseObject.OutputTypeExProperty,
+                    Value = enumValue.Object,
+                };
+
+                var projectProperties = ProjectPropertiesFactory.Create(project, data);
+                var activeConfiguredProject = ActiveConfiguredProjectFactory.ImplementValue(() => projectProperties);
+
+                var vsLangProjectProperties = new VsLangProjectProperties(Mock.Of<VSProject>(), IProjectThreadingServiceFactory.Create(), activeConfiguredProject);
+                Assert.Equal(vsLangProjectProperties.OutputTypeEx, prjOutputTypeEx.prjOutputTypeEx_Library);
+            }
+
+            [Fact]
+            public void VsLangProjectProperties_OutputType_Get()
```

Comment:
I want to mock the call `SetValueAsync` of the `OutputType` property.

`SetupProperty` can be used to mock properties that are overrideable. In this case though, the property will be OutputType, from `ConfigurationGeneralBrowseObject` , which is not overrideable. So it cannot be mocked.

I tried something like this:

```C#
        [Fact]
        public void VsLangProjectProperties_AssemblyName_Get()
        {
            var project = UnconfiguredProjectFactory.Create();
            var data = new PropertyPageData()
            {
                Category = ConfigurationGeneral.SchemaName,
                PropertyName = ConfigurationGeneral.AssemblyNameProperty,
                Value = "Blah"
            };

            var projectProperties = ProjectPropertiesFactory.Create(project, data);
            var activeConfiguredProject = ActiveConfiguredProjectFactory.ImplementValue(() => projectProperties);

            var vsLangProjectProperties = new VSProject(Mock.Of<VSLangProj.VSProject>(), IProjectThreadingServiceFactory.Create(), activeConfiguredProject);
            Assert.Equal(vsLangProjectProperties.AssemblyName, "Blah");

            var mockObject = Mock.Get(projectProperties.GetConfigurationGeneralPropertiesAsync().Result.AssemblyName);
            mockObject.Setup(prop => prop.SetValueAsync(It.IsAny<string>()))
                      .Verifiable();

            vsLangProjectProperties.AssemblyName = "Testing";

            mockObject.Verify();
        }
```

and it results in 
```
       Moq.MockVerificationException : The following setups were not matched:
        IEvaluatedProperty prop => prop.SetValueAsync(It.IsAny<String>())

        Stack Trace:
             at Moq.Mock.Verify()
```